File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/PaymentReconciliation.ts
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Money } from "./Money";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type PaymentReconciliationProcessNote = BackboneElement & {
    text?: string;
    type?: string;
};

export type PaymentReconciliationDetail = BackboneElement & {
    response?: Reference;
    amount?: Money;
    date?: string;
    request?: Reference;
    type: CodeableConcept;
    responsible?: Reference;
    payee?: Reference;
    predecessor?: Identifier;
    identifier?: Identifier;
    submitter?: Reference;
};

export type PaymentReconciliation = DomainResource & {
    requestor?: Reference;
    request?: Reference;
    paymentAmount: Money;
    processNote?: PaymentReconciliationProcessNote[];
    created: string;
    outcome?: string;
    disposition?: string;
    paymentIdentifier?: Identifier;
    status: string;
    paymentDate: string;
    identifier?: Identifier[];
    period?: Period;
    paymentIssuer?: Reference;
    formCode?: CodeableConcept;
    detail?: PaymentReconciliationDetail[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/PaymentReconciliation.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import CodeableConcept
from base import DomainResource
from base import Money
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class PaymentReconciliationProcessNote(BackboneElement):
    text: Optional[str] = None
    type: Optional[str] = None

@dataclass(kw_only=True)
class PaymentReconciliationDetail(BackboneElement):
    response: Optional[Reference] = None
    amount: Optional[Money] = None
    date: Optional[str] = None
    request: Optional[Reference] = None
    type: CodeableConcept
    responsible: Optional[Reference] = None
    payee: Optional[Reference] = None
    predecessor: Optional[Identifier] = None
    identifier: Optional[Identifier] = None
    submitter: Optional[Reference] = None

@dataclass(kw_only=True)
class PaymentReconciliation(DomainResource):
    requestor: Optional[Reference] = None
    request: Optional[Reference] = None
    payment_amount: Money
    process_note: Optional[List[PaymentReconciliationProcessNote]] = field(default_factory=list)
    created: str
    outcome: Optional[str] = None
    disposition: Optional[str] = None
    payment_identifier: Optional[Identifier] = None
    status: str
    payment_date: str
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    period: Optional[Period] = None
    payment_issuer: Optional[Reference] = None
    form_code: Optional[CodeableConcept] = None
    detail: Optional[List[PaymentReconciliationDetail]] = field(default_factory=list)